### PR TITLE
Update to 3.9.1.rc1

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

**Comment by mbargull** [comment]
@isuruf, did you mean to merge this to `master`?

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -10,3 +10,5 @@ cxx_compiler:                  # [win]
   - vs2017                     # [win]
 vc:                            # [win]
   - 14.1                       # [win]
+channel_targets:               # [not (osx and arm64)]
+  - conda-forge python_rc      # [not (osx and arm64)]
```
Comment (mbargull) [inline recipe/conda_build_config.yaml:14]: Why not for `osx and arm64`?

**Comment by isuruf** [comment]
Yes, I meant to merge into master. We can update master when 3.9.1 lands

File: recipe/conda_build_config.yaml (hunk 1 above)
Comment (isuruf) [reply recipe/conda_build_config.yaml:14]: `osx and arm64` is an experimental platform and therefore release candidates are okay there. Need this on arm64.

Comment (mbargull) [reply recipe/conda_build_config.yaml:14]: okay

**Comment by mbargull** [comment]
Generally it'd be problematic because of ABI migrations (thanks for fixing them for this feedstock, btw).
I think it's okay-ish to merge this to `master` given the `osx-arm64`->`label/main` and the final release is not too far off.

**Review by mbargull — APPROVED** [review]

**Comment by isuruf** [comment]
Thanks

**Pull request merged**